File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/receiverQueryApi/ReceiverQueryApiImpl.java
package org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi;

import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.Alerting;
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.AudioAccessibilityPref;
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.BaseURI;
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.CC;
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.CaptionDisplay;
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.Languages;
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.RatingLevel;
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.Service;
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.ServiceGuideUrls;

import java.util.List;

public class ReceiverQueryApiImpl implements IReceiverQueryApi {

    @Override
    public RatingLevel queryContentAdvisoryRating() {
        return null;
    }

    @Override
    public CC queryClosedCaptionsStatus() {
        return null;
    }

    @Override
    public Service queryServiceID() {
        return null;
    }

    @Override
    public Languages queryLanguagePreferences() {
        return null;
    }

    @Override
    public CaptionDisplay queryCaptionDisplayPreferences() {
        return null;
    }

    @Override
    public AudioAccessibilityPref queryAudioAccessibilityPreferences() {
        return null;
    }

    @Override
    public BaseURI queryReceiverWebServerURI() {
        return null;
    }

    @Override
    public List<Alerting> queryAlertingSignaling() {
        return null;
    }

    @Override
    public List<ServiceGuideUrls> queryServiceGuideURLs() {
        return null;
    }
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/receiverQueryApi/ReceiverQueryApiImpl.kt
package org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi

import org.ngbp.jsonrpc4jtestharness.gateway.rpc.IRPCGateway
import org.ngbp.jsonrpc4jtestharness.rpc.receiverQueryApi.model.*

class ReceiverQueryApiImpl(
        private val gateway: IRPCGateway
) : IReceiverQueryApi {

    override fun queryContentAdvisoryRating(): RatingLevel {
        return RatingLevel()
    }

    override fun queryClosedCaptionsStatus(): CC {
        return CC()
    }

    override fun queryServiceID(): Service {
        return Service().apply {
            this.service = gateway.queryServiceId
        }
    }

    override fun queryLanguagePreferences(): Languages {
        return Languages().apply {
            gateway.language.let { language ->
                preferredAudioLang = language
                preferredCaptionSubtitleLang = language
                preferredUiLang = language
            }
        }
    }

    override fun queryCaptionDisplayPreferences(): CaptionDisplay {
        return CaptionDisplay()
    }

    override fun queryAudioAccessibilityPreferences(): AudioAccessibilityPref {
        return AudioAccessibilityPref()
    }

    override fun queryMPDUrl(): MPDUrl {
        return MPDUrl(gateway.mediaUrl)
    }

    override fun queryReceiverWebServerURI(): BaseURI {
        return BaseURI()
    }

    override fun queryAlertingSignaling(): Alerting {
        return Alerting()
    }

    override fun queryServiceGuideURLs(): ServiceGuideUrls {
        return ServiceGuideUrls()
    }
}